### PR TITLE
DevOps GitHub actions pipeline

### PR DESCRIPTION
**Pull request by ManishSihag**
## 🎯 Summary

  
https://citz-imb.atlassian.net/browse/DEVOP-168

The changes allows new pipeline to be implemented for dev envioronment

New workflows files added for the pipeline
- Dev Api Image Build
- Dev APP image Build
- Dev API image Deploy
- Dev APP image Deploy

Helper
- deploy.sh : to process the deployment config
- update wiki table.py - this is the script that update the column of the wiki page according to image build and deploy

Openshift
- new templates added for the deployment at openshift/templates

Dockerfile - Frontend
- Updated dockerfile.ocp to enable nginx

Nginx Prod configuration
- holds the configuration for the nginx component for openshift. 

Existing workflow update to be triggered at main instead of dev
.github/workflows/api-dotnetcore.yml
.github/workflows/app-jest-coverage.yml

## 🔰 Checklist

- [ ] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: .github/helpers/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# Use DeploymentConfigs to deploy the application to OpenShift.
+
+
+    oc process -f /home/runner/work/PIMS/PIMS/openshift/templates/"$DC_TEMPLATE" --namespace=$NAMESPACE \
+        -p LICENSE_PLATE=$LICENSE_PLATE \
+        -p IMAGE_TAG=$IMAGE_TAG \
+        -p ENVIRONMENT=$ENVIRONMENT | \
+        oc apply -f -
```
Comment (akroon3r) [inline .github/helpers/deploy.sh:10]:
```suggestion
        oc apply -f -

```

File: docker-compose.prod.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - 8080:8080
+
+  ################## Database Migration #################
+
+  migration:
+    build:
+      context: backend
+      dockerfile: Dockerfile.migration
+    image: pims-migrations:latest
+
```
Comment (akroon3r) [inline docker-compose.prod.yml:26]:
```suggestion
```

File: openshift/templates/api-dc-template.yaml [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+    value: dev
+  - description: License Plate
+    displayName: License Plate
+    name: LICENSE_PLATE
+    required: true
+    value: ec1236
+  - description: ImageTag
+    displayName: ImageTag
+    name: IMAGE_TAG
+    required: true
```
Comment (akroon3r) [inline openshift/templates/api-dc-template.yaml:227]:
```suggestion
    required: true

```

File: openshift/templates/app-dc-template.yaml [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    value: dev
+  - description: License Plate
+    displayName: License Plate
+    name: LICENSE_PLATE
+    required: true
+    value: ec1236
+  - description: ImageTag
+    displayName: ImageTag
+    name: IMAGE_TAG
+    required: true
```
Comment (akroon3r) [inline openshift/templates/app-dc-template.yaml:155]:
```suggestion
    required: true

```

**Review by akroon3r — CHANGES_REQUESTED** [review]
a couple of small nits with CRLF, otherwise it looks good on first pass

**Review by akroon3r — APPROVED** [review]

File: docker-compose.prod.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    ports:
+      - 8080:8080
+
+  ################## Database Migration #################
+
+  migration:
+    build:
+      context: backend
+      dockerfile: Dockerfile.migration
+    image: pims-migrations:latest
```
Comment (dbarkowsky) [inline docker-compose.prod.yml:25]:
```suggestion
    image: pims-migrations:latest
    
```

**Review by dbarkowsky — COMMENTED** [review]
One more new line needed.

**Review by dbarkowsky — APPROVED** [review]
Looks good. Seeing successful deployment.

**Pull request merged**